app/services/collators/gross_income_collator.rb

```diff
@@ -14,15 +15,21 @@ def populate_attrs
       attrs = default_attrs
 
       INCOME_CATEGORIES.each do |category|
-        attrs[:"#{category}_bank"] = category == :benefits ? monthly_state_benefits : categorised_income[category].to_d
-        attrs[:"#{category}_cash"] = monthly_transaction_amount_by(operation: :credit, category: category)
-        attrs[:"#{category}_all_sources"] = attrs[:"#{category}_bank"] + attrs[:"#{category}_cash"]
-        attrs[:total_gross_income] += attrs[:"#{category}_all_sources"]
+        populate_income_attrs(attrs, category)
       end
+      attrs[:total_gross_income] += gross_income_summary.gross_employment_income
 
       attrs
     end
 
+    def populate_income_attrs(attrs, category)
+      attrs[:"#{category}_bank"] = category == :benefits ? monthly_state_benefits : categorised_income[category].to_d
+      attrs[:"#{category}_cash"] = monthly_transaction_amount_by(operation: :credit, category: category)
+      attrs[:"#{category}_all_sources"] = attrs[:"#{category}_bank"] + attrs[:"#{category}_cash"]
+      attrs[:total_gross_income] += attrs[:"#{category}_all_sources"]
+      attrs
```

Comment:
I'm not sure you need to return `attrs` here, it doesn't look like the return value from this method is being used anywhere. (I could be wrong though, I haven't tried it out).

Reply:
thanks @mpw5 , changed.